File: src/wrappers/python/samples/sign_gl19.py
#!/usr/bin/env python3

from pygroupsig import groupsig
from pygroupsig import signature
from pygroupsig import memkey
from pygroupsig import grpkey
from pygroupsig import mgrkey
from pygroupsig import constants

# Setup
issuer = groupsig.setup(constants.GL19_CODE)
_gpk = issuer['grpkey']
isk = issuer['mgrkey']
converter = groupsig.setup(constants.GL19_CODE, _gpk);
csk = converter['mgrkey']
gpk = converter['grpkey']

# Join
msg1 = groupsig.join_mgr(0, isk, gpk)
msg2 = groupsig.join_mem(1, gpk, msgin = msg1)
usk = msg2['memkey']
msg3 = groupsig.join_mgr(2, isk, gpk, msg2['msgout'])
msg4 = groupsig.join_mem(3, gpk, msgin = msg3, memkey = usk)
usk = msg4['memkey']

# Sign
sig = groupsig.sign("Hello, World!", usk, gpk)

# Verify
b = groupsig.verify(sig, "Hello, World!", gpk)

if b == True:
    print ("VALID signature.")
else:
    print ("WRONG signature.")
    sys.exit()

groupsig.clear(constants.GL19_CODE, issuer['config'])
groupsig.clear(constants.GL19_CODE, converter['config'])

File: src/wrappers/nodejs/samples/sign_gl19.js
#!/usr/bin/env nodejs

const jsgroupsig = require('jsgroupsig');

/* Init groupsig */
jsgroupsig.init(jsgroupsig.GL19);

/* Init grp_key */
let grpkey = jsgroupsig.grp_key_init(jsgroupsig.GL19);

/* Init issuer and converter keys */
let isskey = jsgroupsig.mgr_key_init(jsgroupsig.GL19);
let cnvkey = jsgroupsig.mgr_key_init(jsgroupsig.GL19);

/* Setup call 1: initializes (partial) group key and issuer key */
jsgroupsig.setup(jsgroupsig.GL19, grpkey, isskey);

/* Setup call 2: completes group key and initializes converter key */
jsgroupsig.setup(jsgroupsig.GL19, grpkey, cnvkey);

/* Add a member */
let memkey = jsgroupsig.mem_key_init(jsgroupsig.GL19);
let msg1 = jsgroupsig.join_mgr(0, isskey, grpkey);
let msg2 = jsgroupsig.join_mem(1, memkey, grpkey, msg1);
let msg3 = jsgroupsig.join_mgr(2, isskey, grpkey, msg2);
let msg4 = jsgroupsig.join_mem(3, memkey, grpkey, msg3);

/* sign */
let sig = jsgroupsig.sign("Hello, World!", memkey, grpkey);

/* verify */
let ok = jsgroupsig.verify(sig, "Hello, World!", grpkey);

if (ok) console.log("VALID signature.");
else console.log("WRONG signature.");

jsgroupsig.clear(jsgroupsig.GL19);
